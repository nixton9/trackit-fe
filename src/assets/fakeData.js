export const notes = [
  {
    id: 1,
    title: 'Lista de compras',
    date: '2020-07-20',
    content: 'asdasdad',
    tags: [
      { id: 1, name: 'Primary', color: '#F44336' },
      { id: 2, name: 'Secondary', color: '#E91E63' }
    ]
  },
  {
    id: 2,
    title: 'Apontamentos aula',
    date: '2020-06-25',
    content: 'asdasdad',
    tags: [
      { id: 3, name: 'Studies', color: '#607D8B' },
      { id: 4, name: 'School', color: '#00BCD4' }
    ]
  },
  {
    id: 3,
    title: 'Random shit',
    date: '2020-07-15',
    content: 'asdasdad',
    tags: [{ id: 5, name: 'Random', color: '#3F51B5' }]
  },
  {
    id: 4,
    title: 'Work shit',
    date: '2020-07-15',
    content: 'asdasdad',
    tags: [
      { id: 6, name: 'Work', color: '#9C27B0' },
      { id: 7, name: 'Coding', color: '#FF9800' },
      { id: 8, name: 'Stuff', color: '#4CAF50' }
    ]
  },
  {
    id: 5,
    title: 'Post 129',
    date: '2020-05-07',
    content: 'asdasdad'
  }
]
