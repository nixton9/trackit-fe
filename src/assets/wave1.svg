<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_294" data-name="Caminho 294" d="M-14697.143-11591.146s204.548,93.02,241.44,226.007,101.77,246.085,225.016,271.63,239.81-133.94,286.815-109.888,72.048,136.969,72.048,136.969,18.206,112.27,0,146.063-154.687,113.568-154.687,113.568-145.483,41.1-252.11,41.649-282.684-4.542-332.2-22.729-57.569-16.607-86.319-46.7-74.028-81.864-81.211-123.045-11.661-111.631-8.345-135.894,30.468-240.38,30.94-257.871-.572-130.1,0-136.284S-14697.143-11591.146-14697.143-11591.146Z" transform="translate(14695 11649)" fill="#273150" opacity="0.051"/>
  </g>
</svg>
