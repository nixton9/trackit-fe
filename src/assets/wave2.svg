<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_307" data-name="Caminho 307" d="M-14243.049-11762.149s99.251-140.3,221.355-63.8,71.256,212.9,188.94,256.211,284.029-180.642,371.026-162.149,131.877,179.9,131.877,179.9L-13407-11443.72l-79.384,66.653s-188.631,35.67-195.492,37.041-321.083,20.295-351.518,20.295-209.207-15.854-221.288-20.295-32.708-292.639-32.708-292.639Z" transform="translate(14215 12160.858)" fill="#273150" opacity="0.041"/>
  </g>
</svg>
