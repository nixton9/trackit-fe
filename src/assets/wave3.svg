<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_308" data-name="Caminho 308" d="M1430.13,18293.221s356.522-204.141,537.172-65.336,273.16-352.586,456.213-279.107,399.449,358.18,399.449,358.18,205.3,357.1,208.9,365.451-160.994,153.961-208.9,147.217-546.637-79.807-705.6-97.068-577.556-19.437-656.578-28.982-318.938-34.217-335.6-53.873,280.937-332.744,280.937-332.744Z" transform="translate(-1614 -17637)" fill="#273150" opacity="0.04"/>
  </g>
</svg>
