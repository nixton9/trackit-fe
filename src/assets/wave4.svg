<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="744" height="744" viewBox="0 0 744 744">
  <defs>
    <clipPath id="clip-Prancheta_7">
      <rect width="744" height="744"/>
    </clipPath>
  </defs>
  <g id="Prancheta_7" data-name="Prancheta â€“ 7" clip-path="url(#clip-Prancheta_7)">
    <path id="Caminho_307" data-name="Caminho 307" d="M1449.88,18572s112.775-319.779,271.458-247.187,168.861,179.166,303.627,151.676,170.134-339.822,339.459-310.615,218.6,83.457,218.6,83.457l52.227,227.158v247.463s-162.445,46.264-172.641,44.359-378.379,27.41-395.417,29.01-439.332,0-439.332,0l-251.667-29.01Z" transform="translate(-1613 -17892)" fill="#273150" opacity="0.04"/>
  </g>
</svg>
