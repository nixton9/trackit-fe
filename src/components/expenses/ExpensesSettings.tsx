import React, { useState, useEffect } from 'react'
import Drawer from '../misc/Drawer'
import { TypeEditor } from './TypeEditor'
import { SelectMenu } from '../misc/SelectMenu'
import { NotificationTypes, notificationState } from '../misc/Notification'
import { alertState } from '../misc/Alert'
import { Styled } from '../../styles/Settings.styles'
import { TYPES, EXPENSES } from '../../utils/queries'
import { useToggleElement } from '../../utils/useToggleElement'
import { ReactComponent as SettingsIcon } from '../../assets/icons/settings.svg'
import { ReactComponent as PlusIcon } from '../../assets/icons/plus.svg'
import { ReactComponent as MinusIcon } from '../../assets/icons/minus.svg'
import {
  ExpenseType,
  Currencies,
  currencyOptions
} from '../../utils/ModuleTypes'
import { useLocalStorage } from '../../utils/useLocalStorage'
import { useSetRecoilState, useRecoilState, atom } from 'recoil'
import { gql, useMutation, useQuery } from '@apollo/client'
import { useHistory } from 'react-router-dom'

export const currencyState = atom({
  key: 'currency',
  default: Currencies.EURO
})

const DELETE_TYPE = gql`
  mutation DeleteType($id: ID!) {
    deleteType(id: $id) {
      id_type
    }
  }
`

type ExpensesSettingsProps = {
  types: ExpenseType[]
}

const ExpensesSettings: React.FC<ExpensesSettingsProps> = ({ types }) => {
  const [showTypeEditor, setShowTypeEditor] = useState(false)
  const [activeType, setActiveType] = useState<ExpenseType | null>(null)

  const [, setLSCurrency] = useLocalStorage('currency', Currencies.EURO)
  const [currency, setCurrency] = useRecoilState(currencyState)

  const setNotification = useSetRecoilState(notificationState)
  const setAlert = useSetRecoilState(alertState)

  const history = useHistory()

  const [open, setOpen, overlayEl] = useToggleElement(() =>
    setShowTypeEditor(false)
  )

  const { refetch: refetchTypes } = useQuery(TYPES, {
    fetchPolicy: 'network-only'
  })
  const { refetch: refetchExpenses } = useQuery(EXPENSES)

  const [deleteType] = useMutation(DELETE_TYPE)

  const handleCurrencyChange = (e: any) => {
    setCurrency(e.target.value)
    setLSCurrency(e.target.value)
  }

  const handleTypeClick = (type: ExpenseType) => {
    setActiveType(type)
    setShowTypeEditor(true)
  }

  const handlePlusClick = () => {
    if (showTypeEditor) {
      setShowTypeEditor(false)
    } else {
      setActiveType(null)
      setShowTypeEditor(true)
    }
  }

  const handleDeleteType = (typeId: string | number) => {
    deleteType({
      variables: {
        id: typeId
      }
    })
      .then(res => {
        setNotification({
          text: 'Type was deleted',
          type: NotificationTypes.Success
        })
        refetchTypes()
        refetchExpenses()
      })
      .catch(err =>
        setNotification({
          text: 'There was a problem, please try again',
          type: NotificationTypes.Error
        })
      )
  }

  const handleDeleteTypeConfirm = (typeId: string | number) => {
    setShowTypeEditor(false)
    setAlert({
      text: 'This type will be removed.',
      onConfirm: () => handleDeleteType(typeId)
    })
  }

  useEffect(() => {
    history.push(`#`)

    window.onpopstate = (e: any) => {
      setOpen(false)
    }
  }, [history, setOpen])

  return (
    <>
      <SettingsIcon
        className="settings-icon"
        onClick={() => setOpen(true)}
        data-test-id="expenses-settings-icon"
      />

      <Drawer title="Settings" open={open} overlayRef={overlayEl}>
        <>
          <Styled.SettingsBlock>
            <Styled.SettingsBlock__Label htmlFor="currency">
              Currency
            </Styled.SettingsBlock__Label>
            <Styled.SettingsBlock__Input>
              <SelectMenu
                id="currency"
                value={currency}
                onChange={handleCurrencyChange}
                options={currencyOptions}
              />
            </Styled.SettingsBlock__Input>
          </Styled.SettingsBlock>

          <Styled.SettingsBlock>
            <Styled.SettingsBlock__Label>
              Categories
            </Styled.SettingsBlock__Label>
            <Styled.SettingsBlock__Categories>
              {types &&
                types.map(type => (
                  <Styled.SettingsBlock__Category
                    color={type.color}
                    key={type.id}
                  >
                    <div
                      className="inner"
                      onClick={() => handleTypeClick(type)}
                    >
                      {type.name}
                    </div>
                    <span
                      className="delete"
                      onClick={() => handleDeleteTypeConfirm(type.id)}
                    >
                      +
                    </span>
                  </Styled.SettingsBlock__Category>
                ))}
              <Styled.SettingsBlock__Icon
                className="mbl-click"
                onClick={handlePlusClick}
                data-test-id="expenses-add-icon"
              >
                {showTypeEditor ? <MinusIcon /> : <PlusIcon />}
              </Styled.SettingsBlock__Icon>
            </Styled.SettingsBlock__Categories>
          </Styled.SettingsBlock>

          {showTypeEditor && (
            <TypeEditor
              type={activeType}
              closeEditor={() => setShowTypeEditor(false)}
            />
          )}
        </>
      </Drawer>
    </>
  )
}

export default ExpensesSettings
