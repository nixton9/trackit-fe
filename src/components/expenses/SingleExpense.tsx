import React from 'react'
import { Styled } from '../../styles/Single.styles'
import { ModuleTypes, Expense, Currencies } from '../../utils/ModuleTypes'
import { showCurrencySym } from '../../utils/globalHelpers'
import { activeContentState, isEditState } from '../misc/Add'
import { expenseIdState } from '../../utils/atoms'
import { useSetRecoilState } from 'recoil'

interface SingleExpenseProps extends Expense {
  currency: Currencies
}

const SingleExpense: React.FC<SingleExpenseProps> = ({
  id,
  title,
  value,
  type,
  currency
}) => {
  const setActiveContent = useSetRecoilState(activeContentState)
  const setTaskId = useSetRecoilState(expenseIdState)
  const setIsEdit = useSetRecoilState(isEditState)

  const handleExpenseEdit = () => {
    setActiveContent(ModuleTypes.Expenses)
    setIsEdit(true)
    setTaskId(id.toString())
  }

  return (
    <Styled.SingleExpense__Container onClick={handleExpenseEdit}>
      <Styled.SingleFlex>
        <Styled.SingleTitle>{title ? title : '-'}</Styled.SingleTitle>
        {type && (
          <Styled.SingleCategory color={type.color}>
            {type.name}
          </Styled.SingleCategory>
        )}
        <Styled.SingleExpense__Value>
          {value}
          {currency && showCurrencySym(currency)}
        </Styled.SingleExpense__Value>
      </Styled.SingleFlex>
    </Styled.SingleExpense__Container>
  )
}

export default SingleExpense
