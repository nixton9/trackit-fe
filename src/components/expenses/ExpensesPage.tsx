import React, { useState } from 'react'
import { ExpensesList } from './ExpensesList'
import { ExpensesStats } from './ExpensesStats'
import { currencyState } from './ExpensesSettings'
import { activeContentState } from '../misc/Add'
import { Styled } from '../../styles/Page.styles'
import { Expense } from '../../utils/ModuleTypes'
import { EXPENSES, TYPES } from '../../utils/queries'
import { parseDate, parseDateInverse } from '../../utils/dateHelpers'
import { startOfMonth, endOfMonth, isWithinInterval } from 'date-fns'
import { useQuery } from '@apollo/client'
import { useSetRecoilState, useRecoilValue } from 'recoil'

type ExpensesPageProps = {
  stats?: boolean
}

const ExpensesPage: React.FC<ExpensesPageProps> = ({ stats }) => {
  const setActiveContent = useSetRecoilState(activeContentState)

  const currency = useRecoilValue(currencyState)

  const { loading, error, data } = useQuery(EXPENSES)
  const { data: types } = useQuery(TYPES)

  const [startDate, setStartDate] = useState(startOfMonth(new Date()))
  const [endDate, setEndDate] = useState(endOfMonth(new Date()))

  const expensesDays: [] | string[] = data
    ? Array.from(
        new Set(
          data.expenses.map((expense: Expense) =>
            parseDateInverse(expense.date)
          )
        )
      )
    : []

  const visibleExpensesDay = data
    ? expensesDays
        .filter((day: string) =>
          isWithinInterval(parseDate(day), {
            start: startDate,
            end: endDate
          })
        )
        .sort(
          (a: string, b: string) =>
            new Date(b).getTime() - new Date(a).getTime()
        )
    : []

  const totalExpensesVal = data
    ? data.expenses
        .filter((expense: Expense) =>
          visibleExpensesDay.includes(parseDateInverse(expense.date))
        )
        .reduce((acc: number, obj: Expense) => acc + obj.value, 0)
        .toFixed(2)
    : 0

  return (
    <Styled.PageContainer className="overflow">
      {stats ? (
        <ExpensesStats
          data={data}
          categories={types}
          error={error}
          loading={loading}
        />
      ) : (
        <ExpensesList
          startDate={startDate}
          setStartDate={setStartDate}
          endDate={endDate}
          setEndDate={setEndDate}
          totalExpensesVal={totalExpensesVal}
          visibleExpensesDay={visibleExpensesDay}
          currency={currency}
          data={data}
          types={types}
          error={error}
          loading={loading}
          setActiveContent={setActiveContent}
        />
      )}
    </Styled.PageContainer>
  )
}

export default ExpensesPage
