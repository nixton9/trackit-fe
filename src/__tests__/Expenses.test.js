import React from 'react'
import { render } from '@testing-library/react'
import ExpensesPage from '../components/expenses/ExpensesPage'
import ExpensesSettings from '../components/expenses/ExpensesSettings'
import SingleExpense from '../components/expenses/SingleExpense'
import { expenses, expensesCategories } from '../assets/fakeData'
import { BrowserRouter } from 'react-router-dom'

describe('Expenses Page', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <ExpensesPage />
      </BrowserRouter>
    )

    expect(container).toMatchSnapshot()
  })

  it('has the title', () => {
    const { getByText } = render(
      <BrowserRouter>
        <ExpensesPage />
      </BrowserRouter>
    )

    expect(getByText('Expenses')).toBeInTheDocument()
  })
})

describe('SingleExpense', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <SingleExpense
          id={expenses[0].id}
          title={expenses[0].title}
          date={expenses[0].date}
          value={expenses[0].value}
          category={expenses[0].category}
        />
      </BrowserRouter>
    )

    expect(container).toMatchSnapshot()
  })

  it('has all the content of the task', () => {
    const { getByText } = render(
      <BrowserRouter>
        <SingleExpense
          id={expenses[0].id}
          title={expenses[0].title}
          date={expenses[0].date}
          value={expenses[0].value}
          category={expenses[0].category}
        />
      </BrowserRouter>
    )

    expect(getByText('Conta Hospital')).toBeInTheDocument()
    expect(getByText('57.75$')).toBeInTheDocument()
    expect(getByText('Saúde')).toBeInTheDocument()
  })
})

describe('ExpensesSettings', () => {
  it('matches snashot', () => {
    const { container } = render(
      <ExpensesSettings categories={expensesCategories} />
    )

    expect(container).toMatchSnapshot()
  })

  it('has all the fields', () => {
    const { getByText } = render(
      <ExpensesSettings categories={expensesCategories} />
    )

    expect(getByText('Currency')).toBeInTheDocument()
    expect(getByText('Font size')).toBeInTheDocument()
    expect(getByText('Categories')).toBeInTheDocument()
  })

  it('shows tags', () => {
    const { getByText } = render(
      <ExpensesSettings categories={expensesCategories} />
    )

    expect(getByText('Transportes')).toBeInTheDocument()
    expect(getByText('Saúde')).toBeInTheDocument()
    expect(getByText('Casa')).toBeInTheDocument()
    expect(getByText('Vestuário')).toBeInTheDocument()
    expect(getByText('Educação')).toBeInTheDocument()
    expect(getByText('Lazer')).toBeInTheDocument()
  })
})
