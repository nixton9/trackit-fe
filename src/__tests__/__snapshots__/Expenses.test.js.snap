// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expenses Page matches snapshot 1`] = `
<div>
  <div
    class="sc-fzoLag eydcJC"
  >
    <h1
      class="sc-fzoXzr MtNNd"
    >
      Expenses
    </h1>
    <div
      class="sc-fzpjYC gnMnem"
    >
      <div
        class="sc-fznxsB dsgAhn"
      >
        <div
          class="sc-fznJRM dEkFjf"
        >
          01 Aug
           - 
          31 Aug
          <svg>
            chevron.svg
          </svg>
        </div>
        <h3
          class="sc-fznWqX jXPxyW"
        >
          346.5
          $
        </h3>
      </div>
      <div
        class="sc-fzoiQi dFUsJk"
      >
        <svg
          class="settings-icon"
        >
          settings.svg
        </svg>
        <div
          class="sc-AxjAm khxSQh"
        >
          <h2
            class="sc-AxiKw jzoYkp"
          >
            Settings
          </h2>
          <div
            class="sc-AxhCb iGvJwE"
          >
            <div
              class="sc-AxhUy gnZxSn"
            >
              <label
                class="sc-AxgMl fHkhzP"
                for="currency"
              >
                Currency
              </label>
              <div
                class="sc-AxheI hCWMTo"
              >
                <div
                  class="MuiInputBase-root MuiInput-root"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="currency"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="currency"
                    role="button"
                    tabindex="0"
                  >
                    Euro €
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="euro"
                  />
                  <svg
                    class="MuiSelect-icon"
                  >
                    chevron.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="sc-AxhUy gnZxSn"
            >
              <label
                class="sc-AxgMl fHkhzP"
                for="font-size"
              >
                Font size
              </label>
              <div
                class="sc-AxheI hCWMTo"
              >
                <div
                  class="MuiInputBase-root MuiInput-root"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="font-size"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="font-size"
                    role="button"
                    tabindex="0"
                  >
                    Medium
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="medium"
                  />
                  <svg
                    class="MuiSelect-icon"
                  >
                    chevron.svg
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="sc-AxhUy gnZxSn"
            >
              <label
                class="sc-AxgMl fHkhzP"
              >
                Categories
              </label>
              <div
                class="sc-AxmLO ceYEas"
              >
                <p
                  class="sc-fzozJi ihgPmv"
                  color="#673AB7"
                >
                  Transportes
                </p>
                <p
                  class="sc-fzozJi gqjDvs"
                  color="#FFC107"
                >
                  Saúde
                </p>
                <p
                  class="sc-fzozJi fKpEFM"
                  color="#FF9800"
                >
                  Casa
                </p>
                <p
                  class="sc-fzozJi dNhvgN"
                  color="#9E9E9E"
                >
                  Vestuário
                </p>
                <p
                  class="sc-fzozJi tItWo"
                  color="#E91E63"
                >
                  Educação
                </p>
                <p
                  class="sc-fzozJi bTUfLY"
                  color="#4CAF50"
                >
                  Lazer
                </p>
                <span
                  class="sc-fzoLsD bpeBKU"
                >
                  <svg>
                    plus.svg
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-AxirZ gpexXZ"
        />
      </div>
    </div>
    <section
      class="sc-fzqARJ elDNLU"
    >
      <div
        class="sc-fzqNqU dPyFOh"
      >
        <h4
          class="sc-fzoyTs MYaZa"
        >
          04 Aug
        </h4>
        <h4
          class="sc-fzoNJl gNjhoU"
        >
          <article
            class="sc-fzqBZW dBijkG"
          >
            <div
              class="sc-fzplWN gwxuzL"
            >
              <h5
                class="sc-fznyAO drGMcz"
              >
                Conta Hospital
              </h5>
              <h6
                class="sc-fznZeY ekLkIy"
                color="#FFC107"
              >
                Saúde
              </h6>
              <p
                class="sc-fzqNJr hKWXie"
              >
                57.75
                $
              </p>
            </div>
          </article>
        </h4>
      </div>
      <div
        class="sc-fzqNqU dPyFOh"
      >
        <h4
          class="sc-fzoyTs MYaZa"
        >
          05 Aug
        </h4>
        <h4
          class="sc-fzoNJl gNjhoU"
        >
          <article
            class="sc-fzqBZW dBijkG"
          >
            <div
              class="sc-fzplWN gwxuzL"
            >
              <h5
                class="sc-fznyAO drGMcz"
              >
                Compras Continente
              </h5>
              <h6
                class="sc-fznZeY gooBaK"
                color="#FF9800"
              >
                Casa
              </h6>
              <p
                class="sc-fzqNJr hKWXie"
              >
                57.75
                $
              </p>
            </div>
          </article>
          <article
            class="sc-fzqBZW dBijkG"
          >
            <div
              class="sc-fzplWN gwxuzL"
            >
              <h5
                class="sc-fznyAO drGMcz"
              >
                Gasóleo
              </h5>
              <h6
                class="sc-fznZeY cjOYaZ"
                color="#673AB7"
              >
                Transportes
              </h6>
              <p
                class="sc-fzqNJr hKWXie"
              >
                57.75
                $
              </p>
            </div>
          </article>
        </h4>
      </div>
      <div
        class="sc-fzqNqU dPyFOh"
      >
        <h4
          class="sc-fzoyTs MYaZa"
        >
          07 Aug
        </h4>
        <h4
          class="sc-fzoNJl gNjhoU"
        >
          <article
            class="sc-fzqBZW dBijkG"
          >
            <div
              class="sc-fzplWN gwxuzL"
            >
              <h5
                class="sc-fznyAO drGMcz"
              >
                Compras Pull&Bear
              </h5>
              <h6
                class="sc-fznZeY kvbuor"
                color="#9E9E9E"
              >
                Vestuário
              </h6>
              <p
                class="sc-fzqNJr hKWXie"
              >
                57.75
                $
              </p>
            </div>
          </article>
        </h4>
      </div>
      <div
        class="sc-fzqNqU dPyFOh"
      >
        <h4
          class="sc-fzoyTs MYaZa"
        >
          08 Aug
        </h4>
        <h4
          class="sc-fzoNJl gNjhoU"
        >
          <article
            class="sc-fzqBZW dBijkG"
          >
            <div
              class="sc-fzplWN gwxuzL"
            >
              <h5
                class="sc-fznyAO drGMcz"
              >
                Livro Amazon
              </h5>
              <h6
                class="sc-fznZeY kLrACO"
                color="#E91E63"
              >
                Educação
              </h6>
              <p
                class="sc-fzqNJr hKWXie"
              >
                57.75
                $
              </p>
            </div>
          </article>
        </h4>
      </div>
      <div
        class="sc-fzqNqU dPyFOh"
      >
        <h4
          class="sc-fzoyTs MYaZa"
        >
          10 Aug
        </h4>
        <h4
          class="sc-fzoNJl gNjhoU"
        >
          <article
            class="sc-fzqBZW dBijkG"
          >
            <div
              class="sc-fzplWN gwxuzL"
            >
              <h5
                class="sc-fznyAO drGMcz"
              >
                Jogo XBOX
              </h5>
              <h6
                class="sc-fznZeY djbAQy"
                color="#4CAF50"
              >
                Lazer
              </h6>
              <p
                class="sc-fzqNJr hKWXie"
              >
                57.75
                $
              </p>
            </div>
          </article>
        </h4>
      </div>
    </section>
  </div>
</div>
`;

exports[`ExpensesSettings matches snashot 1`] = `
<div>
  <svg
    class="settings-icon"
  >
    settings.svg
  </svg>
  <div
    class="sc-AxjAm khxSQh"
  >
    <h2
      class="sc-AxiKw jzoYkp"
    >
      Settings
    </h2>
    <div
      class="sc-AxhCb iGvJwE"
    >
      <div
        class="sc-AxhUy gnZxSn"
      >
        <label
          class="sc-AxgMl fHkhzP"
          for="currency"
        >
          Currency
        </label>
        <div
          class="sc-AxheI hCWMTo"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="currency"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="currency"
              role="button"
              tabindex="0"
            >
              Euro €
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="euro"
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy gnZxSn"
      >
        <label
          class="sc-AxgMl fHkhzP"
          for="font-size"
        >
          Font size
        </label>
        <div
          class="sc-AxheI hCWMTo"
        >
          <div
            class="MuiInputBase-root MuiInput-root"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="font-size"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="font-size"
              role="button"
              tabindex="0"
            >
              Medium
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="medium"
            />
            <svg
              class="MuiSelect-icon"
            >
              chevron.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy gnZxSn"
      >
        <label
          class="sc-AxgMl fHkhzP"
        >
          Categories
        </label>
        <div
          class="sc-AxmLO ceYEas"
        >
          <p
            class="sc-fzozJi ihgPmv"
            color="#673AB7"
          >
            Transportes
          </p>
          <p
            class="sc-fzozJi gqjDvs"
            color="#FFC107"
          >
            Saúde
          </p>
          <p
            class="sc-fzozJi fKpEFM"
            color="#FF9800"
          >
            Casa
          </p>
          <p
            class="sc-fzozJi dNhvgN"
            color="#9E9E9E"
          >
            Vestuário
          </p>
          <p
            class="sc-fzozJi tItWo"
            color="#E91E63"
          >
            Educação
          </p>
          <p
            class="sc-fzozJi bTUfLY"
            color="#4CAF50"
          >
            Lazer
          </p>
          <span
            class="sc-fzoLsD bpeBKU"
          >
            <svg>
              plus.svg
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-AxirZ gpexXZ"
  />
</div>
`;

exports[`SingleExpense matches snapshot 1`] = `
<div>
  <article
    class="sc-fzqBZW dBijkG"
  >
    <div
      class="sc-fzplWN gwxuzL"
    >
      <h5
        class="sc-fznyAO drGMcz"
      >
        Conta Hospital
      </h5>
      <h6
        class="sc-fznZeY ekLkIy"
        color="#FFC107"
      >
        Saúde
      </h6>
      <p
        class="sc-fzqNJr hKWXie"
      >
        57.75
        $
      </p>
    </div>
  </article>
</div>
`;
